frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/create-sc.tsx

```diff
@@ -288,47 +262,77 @@ const CreateSC: React.FC<CreateSCProps> = ({ match, hasNoProvSC, mode, lsoNs })
     }
   };
 
-  const makeCall = (activeStep: WizardStep, onNext: OnNextClick) => {
-    // TODO: Need to think of a way to remove this
-    dispatch({ type: 'setOnNextClick', value: onNext });
-    if (activeStep.id === CreateStepsSC.DISCOVER) {
-      makeAutoDiscoveryCall(onNext, state, dispatch, lsoNs);
-    } else if (activeStep.id === CreateStepsSC.STORAGECLASS) {
-      dispatch({ type: 'setShowConfirmModal', value: true });
-    } else if (activeStep.id === CreateStepsSC.REVIEWANDCREATE) {
-      createCluster();
-    } else {
-      onNext();
+  const activeStepActions = (activeStep: WizardStep, onNext: () => void) => {
```

Comment:
```suggestion
  const activeStepFactory = (onNextClick, isNextDisabled) => ({onNextClick, isNextDisabled});
  
  const activeStepActions = (activeStep: WizardStep, onNext: () => void) => {
```
And just use the factory throughout here. WDYT?

Reply:
Sounds good , I would make it inside `WizardContextConsumer` since its not reused anywhere.